File: gocode/bitseal_ws/simpletoken_test.go
package bitsealws

import (
	"encoding/json"
	"testing"

	ec "github.com/bsv-blockchain/go-sdk/primitives/ec"
)

func fixedPriv(b byte) *ec.PrivateKey {
	buf := make([]byte, 32)
	buf[31] = b
	k, _ := ec.PrivateKeyFromBytes(buf)
	return k
}

func TestSimpleTokenRoundtrip(t *testing.T) {
	priv := fixedPriv(3)
	pub := priv.PubKey()
	payload := map[string]any{"foo": "bar"}
	tok, err := CreateToken(payload, priv, 60)
	if err != nil {
		t.Fatal(err)
	}
	claims, err := VerifyToken(tok, pub)
	if err != nil {
		t.Fatal(err)
	}
	if claims["foo"] != "bar" {
		b, _ := json.Marshal(claims)
		t.Fatalf("unexpected claims %s", string(b))
	}
}

File: tscode/bitseal_ws/SimpleToken.test.ts
import { describe, it, expect } from 'vitest'
import PrivateKey from '@bsv/sdk/primitives/PrivateKey'
import { createToken, verifyToken } from './SimpleToken'

function priv(byteVal:number){
  const arr = new Array(32).fill(0); arr[31]=byteVal; return new PrivateKey(arr)
}

describe('SimpleToken', () => {
  it('roundtrip', () => {
    const p = priv(5)
    const token = createToken({foo:'bar'}, p, 120)
    const claims = verifyToken(token, p.toPublicKey())
    expect(claims.foo).toBe('bar')
  })
}) 